Okta.Auth.Sdk/AuthenticationClient.cs

```diff
@@ -57,23 +30,9 @@ public AuthenticationClient(OktaClientConfiguration apiClientConfiguration = nul
         /// </param>
         /// <param name="httpClient">The HTTP client to use for requests to the Okta API.</param>
         /// <param name="logger">The logging interface to use, if any.</param>
-        public AuthenticationClient(OktaClientConfiguration apiClientConfiguration, HttpClient httpClient, ILogger logger = null)
+        public AuthenticationClient(OktaClientConfiguration apiClientConfiguration = null, HttpClient httpClient = null, ILogger logger = null)
+            : base(apiClientConfiguration, httpClient, logger, "okta-auth-dotnet")
         {
-            Configuration = GetConfigurationOrDefault(apiClientConfiguration);
-            OktaClientConfigurationValidator.Validate(Configuration);
-
-            logger = logger ?? NullLogger.Instance;
-
-            var requestExecutor = new DefaultRequestExecutor(Configuration, httpClient, logger);
-            var resourceFactory = new ResourceFactory(this, logger, new AbstractResourceTypeResolverFactory(ResourceTypeHelper.GetAllDefinedTypes(typeof(Resource))));
```

Comment:
Both SDKs (Auth & Okta) define a specific `Resource` class which inherits from the `BaseResource` class. Although the code looks similar for both SDKs, the `typeof(Resource)` code line is different, and each SDK has to use its specific type.

Reply:
I've fixed the way this is passed in so the child implementations pass in their own AbstractResourceTypeResolverFactory